NuoDb.Data.Client/NuoDbDataReader.cs

```diff
@@ -307,14 +310,17 @@ public override bool NextResult()
 
         public override bool Read()
         {
-            if (this.pendingRows == null)
+            // Currently, afterLast can only be false if pendingRows was non-null in InitResultSet();
+            // - but InitResultSet could be changed in the future.
+            if (afterLast || pendingRows == null)
                 return false;
 
             //int maxRows = statement == null ? 0 : statement.MaxRows;
-            int maxRows = 0;
+            int maxRows = 0;        // this local maxRows HIDES this.maxRows
```

Comment:
If this variable is hiding a class field, why not rename it?

Reply:
I regard this as a hack,

It was in the original code - I didn't make this change originally.

I left this highly dubious code for now, because I wanted just to implement the needed feature in this PR - and not get embroiled in cleaning this code up and ensuring the correct behaviour remains.

In a later PR, I will fix this code by removing the hiding, and cleaning up the code that relies on the hidden var.

To answer you question, there is a condition later in this method that tests maxRows, so hiding maxRows with a local zero (0) value effectively disables that test.
Which is the intention of this code. 

I think the long term solution is to just make sure this class works correctly in the absence of a reliable maxRows value.